catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/RecyclerViewFragment.java

```diff
@@ -461,8 +473,22 @@ protected void renameItem(T item, String newName) {
 		finishActionMode();
 	}
 
-	protected void mergeProjects(List<T> selectedProjects, String mergeProjectName) {
-		ToastUtil.showSuccess(getContext(), R.string.merging_project_text);
+	protected void mergeProjects(List<T> selectedProjects, String mergeProjectName) throws IOException, LoadingProjectException {
+		File firstProjectDir = new File(FlavoredConstants.DEFAULT_ROOT_DIRECTORY, selectedProjects.get(0).getName());
+		File secondProjectDir = new File(FlavoredConstants.DEFAULT_ROOT_DIRECTORY, selectedProjects.get(1).getName());
+		if (firstProjectDir.exists() && firstProjectDir.isDirectory() && secondProjectDir.exists() && secondProjectDir.isDirectory()) {
+			Project firstProject = XstreamSerializer.getInstance().loadProject(firstProjectDir,
+					getContext());
+			Project secondProject = XstreamSerializer.getInstance().loadProject(secondProjectDir,
+					getContext());
+			Project mergedProject = new MergeProject(firstProject, secondProject, getContext(), false).performMerge(mergeProjectName);
```

Comment:
the name of the class "MergeProject" implies to me that it is a Project, a subclass of it. however thats not the case here.